### PR TITLE
[Tests] Add easing functions test.

### PR DESCRIPTION
**Pull request by Faless**
Tests results of easing functions are compared to some known results.
New expected values can be printed to stdout by changing the `#if 0`.

To confirm that #44197 did not introduce regression, I created this simple regression test to compare the results before and after the patch (confirming it did not introduce a regression).

### CODE REVIEW

File: tests/test_easing.h [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+			printf("\t\t{");
+			for (int z = 0; z <= STEPS; z++) {
+				printf("%f,", results[x][y][z]);
+			}
+			printf("},\n");
+		}
+		printf("\t},\n");
+	}
+	printf("\n};\n");
+#endif
```
Comment (akien-mga) [inline tests/test_easing.h:150]: Is it relevant to keep the `#if 0`'ed code?

Comment (Faless) [reply tests/test_easing.h:150]: well, if we ever need to regenerate the array.
But it could also easily be rewritten, so I can remove it if you prefer.

Comment (akien-mga) [reply tests/test_easing.h:150]: Well I guess it doesn't hurt to keep, maybe just improve the comment to explain its purpose (i.e to re-create the results array if there's ever a change to the behavior of Tween and we want to update the test).

**Comment by Xrayez** [comment]
In theory the test data could be saved into a file as well, as done similarly in #44296.

**Comment by akien-mga** [comment]
Might need a rebase, but otherwise is this ready to merge?

File: tests/test_easing.h [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+};
+
+const float TIME_START = 0;
+const float TIME_END = 1;
+const int STEPS = 30;
+
+/* clang-format off */
+const float expected[Tween::TRANS_COUNT][Tween::EASE_COUNT][STEPS+1] = {
+	{
+		{300.000000,316.666656,333.333344,350.000000,366.666687,383.333344,400.000000,416.666687,433.333344,450.000000,466.666687,483.333344,500.000000,516.666687,533.333374,550.000000,566.666687,583.333374,600.000000,616.666687,633.333374,650.000000,666.666687,683.333374,700.000000,716.666687,733.333374,750.000000,766.666687,783.333374,800.000000,},
```
Comment (Calinou) [inline tests/test_easing.h:52]: This is pretty minor, but I'd still separate each value in the array with a space after the comma.

File: tests/test_easing.h [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+				const float t = TIME_START + (TIME_END - TIME_START) / STEPS * i;
+				real_t v = tween.run_equation(tt, et, t, 300, 500, TIME_END);
+				results[tr][ease][i] = v;
+			}
+		}
+	}
+#if 0
+	// Print the results array.
+	// This can be used to re-create the "expected" array in case we know the test
+	// will procuce a different result, and we want to update the test itself.
```
Comment (Calinou) [inline tests/test_easing.h:138]:
```suggestion
	// will produce a different result, and we want to update the test itself.
```

File: tests/test_easing.h [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+	// (e.g. changing steps, values, known behavior, etc).
+	printf("{\n");
+	for (int x = 0; x < Tween::TRANS_COUNT; x++) {
+		printf("\t{\n");
+		for (int y = 0; y < Tween::EASE_COUNT; y++) {
+			printf("\t\t{");
+			for (int z = 0; z <= STEPS; z++) {
+				printf("%f,", results[x][y][z]);
```
Comment (Calinou) [inline tests/test_easing.h:146]: See above:

```suggestion
				printf("%f, ", results[x][y][z]);
```

**Review by Calinou — CHANGES_REQUESTED** [review]

**Comment by Faless** [comment]
I think this is not really needed. Closing.

**Pull request closed without merging**